File: my_solutions/2213.longest-substring-of-one-repeating-character/solution.java
class Node {
    int l, r;
    int lmx, rmx, mx;

    Node(int l, int r) {
        this.l = l;
        this.r = r;
        lmx = rmx = mx = 1;
    }
}

class SegmentTree {
    private char[] s;
    private Node[] tr;

    public SegmentTree(char[] s) {
        int n = s.length;
        this.s = s;
        tr = new Node[n << 2];
        build(1, 1, n);
    }

    public void build(int u, int l, int r) {
        // TODO: implement
    }
}

File: my_solutions/2213.longest-substring-of-one-repeating-character/solution.cpp
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

class Node {
public:
    int l, r;
    int lmx, rmx, mx;

    Node(int l, int r)
        : l(l)
        , r(r)
        , lmx(1)
        , rmx(1)
        , mx(1) {
            // TODO: implement
        }
};
